### PR TITLE
mlx5: Enable enhanced flow steering functionality over the DV API

### PR DESCRIPTION
**Pull request by yishaih**
This series enables enhanced flow steering functionality over the DV API as follows:
- Exposes a device TIR ICM address for RAW and RSS QPs.
- Adds support for SW ICM DM allocation types.
- Allows creating a matcher for a FDB flow table.

The matching kernel part was already sent to rdma-next.

### CODE REVIEW

File: providers/mlx5/libmlx5.map [hunk 1]
```diff
@@ -83,6 +83,7 @@ MLX5_1.9 {
 
 MLX5_1.10 {
 	global:
+		mlx5dv_alloc_dm;
```
Comment (jgunthorpe) [inline providers/mlx5/libmlx5.map]: Is this the right symbol version? Didn't we already release 1.10 ?

File: providers/mlx5/man/mlx5dv_alloc_dm.3.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+**mlx5dv_alloc_dm()** allocates device memory (DM) with specific driver properties.
+
+# ARGUMENTS
+
+Please see *ibv_alloc_dm(3)* man page for *context* and *dm_attr*.
+
+## mlx5_dm_attr
+
+```c
```
Comment (jgunthorpe) [inline providers/mlx5/man/mlx5dv_alloc_dm.3.md]: man pages shouldn't have structs

File: providers/mlx5/man/mlx5dv_alloc_dm.3.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+struct mlx5dv_alloc_dm_attr {
+	enum mlx5dv_alloc_dm_type type;
+	uint64_t comp_mask;
+};
+```
+
+*type*
+:	The device memory type user wishes to allocate:
+
+	MLX5DV_DM_TYPE_MEMIC:
```
Comment (jgunthorpe) [inline providers/mlx5/man/mlx5dv_alloc_dm.3.md]: This doesn't seem to be the usual markdown layout

File: providers/mlx5/man/mlx5dv_create_flow_matcher.3.md [hunk 4]
```diff
@@ -33,13 +33,19 @@ Please see *ibv_open_device(3)* for *context*.
 ## *attr*
 
 ```c
+
+enum mlx5dv_flow_matcher_attr_mask {
+	MLX5DV_FLOW_MATCHER_MASK_FT_TYPE = 1 << 0,
+};
```
Comment (jgunthorpe) [inline providers/mlx5/man/mlx5dv_create_flow_matcher.3.md]: more sturcts in man pages, grrr

File: providers/mlx5/man/mlx5dv_create_flow_matcher.3.md [hunk 5]
```diff
@@ -33,13 +33,19 @@ Please see *ibv_open_device(3)* for *context*.
...
+
 struct mlx5dv_flow_matcher_attr {
 	enum ibv_flow_attr_type type;
 	uint32_t flags; /* From enum ibv_flow_flags */
 	uint16_t priority;
 	uint8_t match_criteria_enable; /* Device spec format */
 	struct mlx5dv_flow_match_parameters *match_mask;
-	uint64_t comp_mask;
+	uint64_t comp_mask; /* use mlx5dv_flow_matcher_attr_mask */
+	enum mlx5dv_flow_table_type ft_type;
```
Comment (jgunthorpe) [inline providers/mlx5/man/mlx5dv_create_flow_matcher.3.md]: And this nonsense with comments after structs is not how we write man pages now, send a fix please.

Comment (yishaih) [reply providers/mlx5/man/mlx5dv_create_flow_matcher.3.md]: man page notes were handled as part of updating the series with the accepted kernel headers.

**Pull request merged**